src/cpp/session/modules/SessionRmdNotebook.R

```diff
@@ -576,13 +370,17 @@ assign(".rs.notebookVersion", envir = .rs.toolsEnv(), "1.0")
       outputFile <- .rs.withChangedExtension(inputFile, ext = ".nb.html")
    
    # generate format
-   format <- .rs.rnb.htmlNotebook()
+   outputFormat <- rmarkdown::html_notebook()
    
    # augment hooks
-   format <- .rs.rnb.cacheAugmentKnitrHooks(rnbData, format)
+   outputFormat <- .rs.rnb.cacheAugmentKnitrHooks(rnbData, outputFormat)
```

Comment:
@jjallaire this still works by taking the notebook format as produced by `rmarkdown`, and augmenting the hooks as needed. Do you think this is okay for the RStudio side? (It also implies that, at least for now, any other front-end would have to do something similar)

I think that _only_ overriding the evaluate hook will not be sufficient as `knitr` on its own does not provide enough context information -- note that we use the `text` and `chunk` hooks to track e.g. line numbers so we can map those back to cached outputs, and so on.